tests/MSBuildDeviceIntegration/Tests/InstallAndRunTests.cs

```diff
@@ -566,16 +561,22 @@ public void RunWithInterpreterEnabled ([Values (false, true)] bool isRelease)
 
 			ClearAdbLogcat ();
 			RunAdbCommand ("shell setprop debug.mono.log all");
+			var logProp = RunAdbCommand ("shell getprop debug.mono.log")?.Trim ();
+			Assert.AreEqual (logProp, "all", "The debug.mono.log prop was not set correctly.");
+
 			if (CommercialBuildAvailable)
 				Assert.True (builder.RunTarget (proj, "_Run"), "Project should have run.");
 			else
 				AdbStartActivity ($"{proj.PackageName}/{proj.JavaPackageName}.MainActivity");
 
-			var logcatFilePath = Path.Combine (Root, builder.ProjectDirectory, "logcat.log");
-			var didStart = WaitForActivityToStart (proj.PackageName, "MainActivity", logcatFilePath, 30);
-			RunAdbCommand ("shell setprop debug.mono.log \"\"");
+			var didStart = WaitForActivityToStart (proj.PackageName, "MainActivity", "startup-logcat.log", 120);
+			RunAdbCommand ("shell setprop debug.mono.log \"''\"");
+			logProp = RunAdbCommand ("shell getprop debug.mono.log")?.Trim ();
+			Assert.AreEqual (logProp, string.Empty, "The debug.mono.log prop was not unset correctly.");
 			Assert.True (didStart, "Activity should have started.");
-			Assert.IsTrue (File.ReadAllText (logcatFilePath).Contains ("Enabling Mono Interpreter"), "logcat output did not contain 'Enabling Mono Interpreter'.");
+			Assert.True (MonitorAdbLogcat ((line) => {
```

Comment:
Style: `line` doesn't *need* the extra parenthesis; `line => …` is fine.

Reply:
For my understanding, is the reason this assert was frequently failing because the test didn't wait long enough between `RunAdbCommand()` and the `File.ReadAllText()` for the "Enabling Mono Interpreter" message to *reliably* exist in the log file?

If that's the case, this appears to be the only instance of this erroneous pattern; all other matches of:

```
% git grep 'File.ReadAllText.*logcat'
build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/RenameTestCases.cs:                               var logcat      = File.ReadAllText (logcatPath);
tests/MSBuildDeviceIntegration/Tests/InstallAndRunTests.cs:                     Assert.IsTrue (ObfuscatedStackRegex.IsMatch (File.ReadAllText (logcatFilePath)), "Original logcat output did not contain obfuscated crash info.");
tests/MSBuildDeviceIntegration/Tests/InstallAndRunTests.cs:                     var logcatOutput = File.ReadAllText (logcatPath);
tests/MSBuildDeviceIntegration/Tests/InstallAndRunTests.cs:                     Assert.IsTrue (File.ReadAllText (logcatFilePath).Contains ("Enabling Mono Interpreter"), "logcat output did not contain 'Enabling Mono Interpreter'.");
```

appear to occur *after* a `MonitorAdbLogcat()` invocation.

Reply:
I am not entirely sure what the root cause was, in some cases I think there was a timing issue in reading the file, or a timing issue in app launching. I could never reproduce this one locally and only saw it on CI.